### PR TITLE
Cli general usage improvements

### PR DESCRIPTION
**Pull request by joacoc**
### Motivation

Has a big dependency on https://github.com/MaterializeInc/materialize/pull/13977

Improvements:

- [x] Add support for `MZ_PROFILE environment var`
- [x] Regions are args
- [x] `--interactive` in login is now a flag
- [x] Args like `eu-west-1` now are `aws/eu-west-1`
- [x] Removed listing the regions when none is provided to `mz shell`
- [x] Removed unused code to disable regions
- [x] Refactor documentation comments

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way) and therefore is tagged with a `T-protobuf` label.
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):

  -

### CODE REVIEW

File: src/mz/src/utils.rs [hunk 1]
```diff
@@ -13,9 +13,66 @@ use indicatif::{ProgressBar, ProgressStyle};
...
+            "aws/eu-west-1" => "eu-west-1",
+            _ => panic!("Unknown region."),
+        }
+    }
+    pub fn parse_enum_region(region: CloudProviderRegion) -> &'static str {
+        match region {
+            CloudProviderRegion::AwsUsEast1 => "us-east-1",
+            CloudProviderRegion::AwsEuWest1 => "eu-west-1",
+        }
+    }
```
Comment (antifuchs) [inline src/mz/src/utils.rs:49]: These methods seem a bit clunky to use. Try this instead?

```suggestion
    /// Return the region name inside a cloud provider.
    pub fn region_name(&self) -> &'static str {
        match self {
            CloudProviderRegion::AwsUsEast1 => "us-east-1",
            CloudProviderRegion::AwsUsEast1 => "eu-west-1",
        }
    }
    
    /// Return the cloud provider name.
    pub fn cloud_provider(self) -> &'static str {
        match self {
            CloudProviderRegion::AwsUsEast1 |  CloudProviderRegion::AwsEuWest1 => "aws",
        }
    }
```

File: src/mz/src/utils.rs [hunk 2]
```diff
@@ -13,9 +13,66 @@ use indicatif::{ProgressBar, ProgressStyle};
...
+/// Implementation to name the possible values and parse every option.
+impl CloudProviderRegion {
+    pub fn variants() -> [&'static str; 2] {
+        ["aws/us-east-1", "aws/eu-west-1"]
+    }
+    pub fn parse(region: String) -> CloudProviderRegion {
+        match region.as_str() {
+            "aws/us-east-1" => CloudProviderRegion::AwsUsEast1,
+            "aws/eu-west-1" => CloudProviderRegion::AwsEuWest1,
+            _ => panic!("Unknown region."),
```
Comment (antifuchs) [inline src/mz/src/utils.rs:34]: This method would be much better as [a FromStr impl](http://doc.rust-lang.org/1.63.0/std/str/trait.FromStr.html) (and then you can remove the `panic` call, which makes me a tiny bit nervous).

**Review by antifuchs — COMMENTED** [review]
Didn't read too deeply, but I found a few things that could be improved. The rest looks fine to me!

File: src/mz/src/login.rs [hunk 3]
```diff
@@ -116,8 +116,8 @@ async fn generate_api_token(
 
 async fn authenticate_user(
     client: &Client,
-    email: String,
-    password: String,
+    email: &String,
```
Comment (mjibson) [inline src/mz/src/login.rs:119]: These can all be `&str`.

File: src/mz/src/main.rs [hunk 4]
```diff
@@ -195,18 +182,22 @@ const PROFILE_NOT_FOUND_MESSAGE: &str =
 #[tokio::main]
 async fn main() {
     let args = Cli::parse();
-    let profile_arg: Option<String> = args.profile;
+    let mut profile_name = DEFAULT_PROFILE_NAME.to_string();
+    let preferedined_profile_arg: Option<String> = args.profile;
+    let predefined_profile_env: Option<&str> = option_env!("MZ_PROFILE");
+
+    if let Some(profile_env) = predefined_profile_env {
```
Comment (mjibson) [inline src/mz/src/main.rs:189]: Clap supports both defaults and env vars, so this should be removed and the appropriate clap options configured.

File: src/mz/src/regions.rs [hunk 5]
```diff
@@ -146,17 +124,16 @@ pub(crate) async fn list_regions(
...
+///
+/// E.g.: [us-east-1, eu-west-1]
 pub(crate) async fn list_cloud_providers(
     client: &Client,
     frontegg_auth_machine: &FronteggAuthMachine,
 ) -> Result<Vec<CloudProvider>, Error> {
     let authorization: String = format!("Bearer {}", frontegg_auth_machine.access_token);
 
-    let headers = build_region_request_headers(authorization);
+    let headers = build_region_request_headers(authorization.as_str());
```
Comment (mjibson) [inline src/mz/src/regions.rs:136]: Should be able to use `&authorization` for these if the funcs are changed to take a `&str`.

File: src/mz/src/utils.rs [hunk 6]
```diff
@@ -13,9 +13,66 @@ use indicatif::{ProgressBar, ProgressStyle};
...
+        }
+    }
+    pub fn parse_region(region: String) -> &'static str {
```
Comment (mjibson) [inline src/mz/src/utils.rs:37]: take a `&str`

**Review by mjibson — APPROVED** [review]

**Pull request merged**